src/test/java/org/biouno/unochoice/UiAcceptanceTest.java

```diff
@@ -84,123 +87,162 @@ public void test() throws Exception {
 
         param3Choices.get(0).click();
-        param3AChoices = driver.findElements(By.cssSelector("div.active-choice:has([name='name'][value='PARAM3A']) input[type='checkbox']"));
+        waitForAsyncCode();
+        param3AChoices = driver.findElements(findCheckbox("PARAM3A"));
         checkRadios(param3AChoices, "A1", "C1", "A2", "C2", "A3", "C3");
 
         param3Choices.get(2).click();
-        param3AChoices = driver.findElements(By.cssSelector("div.active-choice:has([name='name'][value='PARAM3A']) input[type='checkbox']"));
+        waitForAsyncCode();
+        param3AChoices = driver.findElements(findCheckbox("PARAM3A"));
         checkRadios(param3AChoices, "A1", "A2", "A3");
 
         param3Choices.get(0).click();
-        param3AChoices = driver.findElements(By.cssSelector("div.active-choice:has([name='name'][value='PARAM3A']) input[type='checkbox']"));
+        waitForAsyncCode();
+        param3AChoices = driver.findElements(findCheckbox("PARAM3A"));
         checkRadios(param3AChoices);
 
         /*
          PARAM4 and PARAM4A are MULTI_SELECT based
          */
-        WebElement param4Input = driver.findElement(By.cssSelector("div.active-choice:has([name='name'][value='PARAM4']) select"));
-        WebElement param4AInput = driver.findElement(By.cssSelector("div.active-choice:has([name='name'][value='PARAM4A']) select"));
+        WebElement param4Input = driver.findElement(findElement("PARAM4"));
+        WebElement param4AInput = driver.findElement(findElement("PARAM4A"));
 
         checkOptions(param4Input, "A", "B", "C");
         // this is until something is selected
         checkOptions(param4AInput);
 
         new Select(param4Input).selectByVisibleText("A");
+        waitForAsyncCode();
         checkOptions(param4AInput, "A1", "A2", "A3");
 
         new Select(param4Input).selectByVisibleText("B");
+        waitForAsyncCode();
         checkOptions(param4AInput, "A1", "B1", "A2", "B2", "A3", "B3");
 
         new Select(param4Input).selectByVisibleText("C");
+        waitForAsyncCode();
         checkOptions(param4AInput, "A1", "B1", "C1", "A2", "B2", "C2", "A3", "B3", "C3");
 
         new Select(param4Input).deselectByVisibleText("A");
+        waitForAsyncCode();
         checkOptions(param4AInput, "B1", "C1", "B2", "C2", "B3", "C3");
 
         new Select(param4Input).deselectByVisibleText("B");
+        waitForAsyncCode();
         checkOptions(param4AInput, "C1", "C2", "C3");
 
         new Select(param4Input).selectByValue("A");
+        waitForAsyncCode();
         checkOptions(param4AInput, "A1", "C1", "A2", "C2", "A3", "C3");
 
         new Select(param4Input).deselectByValue("C");
+        waitForAsyncCode();
         checkOptions(param4AInput, "A1", "A2", "A3");
 
         new Select(param4Input).deselectByValue("A");
+        waitForAsyncCode();
         checkOptions(param4AInput);
 
     }
 
+    private void waitForAsyncCode() {
+        new WebDriverWait(driver, Duration.ofMinutes(1))
```

Comment:
Is this configurable? In Cypress the timeouts are configured via a configuration that you can tweak when modify without having to touch the sources/tests.

Reply:
I can move it out to a system property or environment variable.

Reply:
You can now pass `-Dui.loading.timeout=PT10S` for example to drop the timeout to 10 seconds. It uses the standard notation used by [`java.time.Duration.parse`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-)